File: FrontEnd/src/app/services/models/devis.ts
/* tslint:disable */
/* eslint-disable */
import {MaterielBattery} from '../models/materiel-battery';
import {MaterielCable} from '../models/materiel-cable';
import {MaterielInverter} from '../models/materiel-inverter';
import {MaterielMeter} from '../models/materiel-meter';
import {MaterielSolarPanel} from '../models/materiel-solar-panel';
import {MaterielSystemFixing} from '../models/materiel-system-fixing';

export interface Devis {
  battery: MaterielBattery;
  cable: MaterielCable;
  construction: { id: string };
  date: string;
  devisRequest: { id: string };
  etat: string;
  id: number;
  inverter: MaterielInverter;
  meter: MaterielMeter;
  ref: string;
  solarPanel: MaterielSolarPanel;
  status: string;
  systemFixing: MaterielSystemFixing;
  total: number;
}

File: BackEnd/src/main/java/com/example/demo/models/Devis.java
package com.example.demo.models;

import com.example.demo.embedded.*;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Entity
@Table(name = "devis")
@Data
public class Devis {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private Date date;
    private String ref;
    private String etat;
    private String status;
    @Embedded
    private MaterielSolarPanel solarPanel;
    @Embedded
    private MaterielInverter inverter;
    @Embedded
    private MaterielSystemFixing systemFixing;
    @Embedded
    private MaterielCable cable;
    @Embedded
    private MaterielBattery battery;
    @Embedded
    private MaterielMeter meter;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @OneToOne
    private DevisRequest devisRequest;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Construction construction;
    private Float total;
}
